geps/gep-713/index.md

```diff
@@ -1,1157 +1,747 @@
 
-Minutes pass while both look at logs.
+### Hierarchy of target kinds and mechanics for calculating Effective meta resources
 
-**Chihiro**: _I’m an idiot. There’s a RetryPolicy for the whole namespace –
-sorry, too many policies in the dashboard and I missed it. Deleting that since
-you don’t want retries._
+Target kinds MUST be organized in a well-known hierarchy of target kinds, from the least specific declared target kinds to the most specific effective ones. This hierarchy shall thus induce a logical representation of the relationships between instances of target objects, in the form of a Directed Acyclic Graph (DAG) whose roots are the least specific objects and the leaves are the most specific ones (and ultimately the effective targets of the meta resources).
 
-**Ana**: _Are you sure that’s a good–_
+The hierarchy of target kinds, as well as the logical representation of hierarchical relationships between instances of target objects, serve as a map for orderly resolving meta resource specs for each combinatorial context that is ultimately affected by the meta resources (i.e. for each Effective target). This process corresponds to resolving the intended augmented behavior that is collectively described by the set of meta resource specs in the context. The spec that summarizes the intended augmented behavior for a given context is referred to as the *Effective meta resource* (or *Effective policy*).
 
-Ana’s phone shrills while she’s typing, and she drops it. When she picks it
-up again she sees a stack of alerts. She goes pale as she quickly flips
-through them: there’s one for every single service in the `baker` namespace.
+The process of calculating Effective meta resources (Effective policies) consists of walking the hierarchy of target objects, from most specific to least specific (i.e., bottom-up, from the leaves towards the roots of the DAG of target objects) or from least specific to most specific (top-down), map reducing to a single meta resource spec each pair of meta resources adjacent to each other in the hierarchy, applying at each step one of the supported merge strategies described below, until no more than one spec remains for each effective target.
 
-**Ana**: _PUT IT BACK!!_
+Meta resource kinds that implement more than one merge strategy MUST provide fields for the instances of the meta resource to specify a chosen strategy (described in the next section). The least specific meta resource of the pair of meta resources whose specs are merged into one dictates the merge strategy to apply in such cases.
 
-**Chihiro**: _Just did. Be glad you couldn't hear all the alarms here._ 😕
+#### Conflict resolution
 
-**Ana**: _What the hell just happened??_
+If multiple meta resources target the same context, this is considered to be a conflict.
```

Comment:
We need to define "context"  again here, I think. (I'd forgotten the definition by the time I got to this part).
```suggestion
If multiple meta resources target the same context (that is, multiple instances of the same or similar policies acting on the same hierarchy have an effective target of the same object), this is considered to be a conflict.
```

Reply:
"same", yes; "similar", not a good idea IMO. I think the behavior for different kinds of policies should be undefined.

Reply:
I'm okay with removing "or similar", but I think that if we're going to leave this as undefined in some cases, we need to be specific in the ones where we _do_ need to have opinions:

* For Gateway API Policy objects included in the specification, in the case of intent conflict with some other Policy on Gateway API objects, the Gateway API Policy must take precedence.
* For implementation specific Policy objects that affect the same properties across multiple implementations, it's up to the implementations to define behavior. If they don't then the behavior is, necessarily, undefined and could produce differing outcomes depending on unknown factors.

In other words, this is a terrible idea and users should try not to use multiple Policy objects that affect the same things.